projects/ad2s1210_iio/app/ad2s1210_iio.c

```diff
@@ -0,0 +1,619 @@
+#include "common.h"
+#include "no_os_error.h"
+#include "no_os_util.h"
+
+/******** Forward declaration of getter/setter functions ********/
+static int iio_ad2s1210_attr_get(void *device, char *buf, uint32_t len,
+				 const struct iio_ch_info *channel,
+				 intptr_t priv);
+
+static int iio_ad2s1210_attr_set(void *device, char *buf, uint32_t len,
+				 const struct iio_ch_info *channel,
+				 intptr_t priv);
+
+/******************************************************************************/
+/********************* Macros and Constants Definition ************************/
+/******************************************************************************/
+#define AD2S1210_CHN_ATTR(_name, _priv) {\
+	.name = _name,\
+	.priv = _priv,\
+	.show = iio_ad2s1210_attr_get,\
+	.store = iio_ad2s1210_attr_set\
+}
+
+#define AD2S1210_CH(_name, _idx, _type, _ch_out) {\
+	.name = _name, \
+	.ch_type = _type,\
+	.ch_out = _ch_out,\
+	.indexed = true,\
+	.channel = 0,\
+	.scan_index = _idx,\
+	.scan_type = &chn_scan[_idx],\
+	.attributes = &ad2s1210_iio_ch_attributes[_idx][0]\
+}
+
+/*	Number of IIO devices */
+#define NUM_OF_IIO_DEVICES	1
+
+#define	BYTES_PER_SAMPLE	2
+
+/* Number of data storage bits (needed for IIO client to plot RESOLVER data) */
+#define CHN_STORAGE_BITS	(BYTES_PER_SAMPLE * 8)
+
+#define AD2S1210_IIO_TRIGGER_NAME         "ad2s1210_iio_trigger"
+
+/* data buffer size */
+#if defined(USE_SDRAM)
+#define data_buffer				SDRAM_START_ADDRESS
+#define DATA_BUFFER_SIZE			SDRAM_SIZE_BYTES
+#else
+#define DATA_BUFFER_SIZE			(32768)		/* 32kbytes */
+static int8_t data_buffer[DATA_BUFFER_SIZE] = { 0 };
+#endif
+
+#define swab16(x) \
+        ((((x) & 0x00ff) << 8) | \
+         (((x) & 0xff00) >> 8))
+
+#define AD2S1210_ENABLE_HYSTERESIS	NO_OS_BIT(4)
+
+enum ad2s1210_attribute_id {
+	RAW_ATTR_ID,
+	SCALE_ATTR_ID,
+	SAMPLING_FREQ_ATTR_ID,
+	LABEL_ATTR_ID,
+	HYSTERESIS_ATTR_ID,
+	HYSTERESIS_AVAILABLE_ATTR_ID,
+	FREQ_ATTR_ID,
+	FREQ_AVAIL_ATTR_ID,
+};
+
+/******************************************************************************/
+/******************** Variables and User Defined Data Types *******************/
+/******************************************************************************/
+
+/* Pointer to the struct representing the AD2S1210 IIO device */
+struct ad2s1210_dev *ad2s1210_dev_inst;
+
+/* IIO interface descriptor */
+static struct iio_desc *ad2s1210_iio_desc;
+
+#if (DATA_CAPTURE_MODE == CONTINUOUS_DATA_CAPTURE)
+static struct iio_hw_trig *ad2s1210_hw_trig_desc;
+#endif
+
+struct scan_type chn_scan[RESOLVER_CHANNELS] = {
+	{
+		.sign = 'u',
+		.realbits = CHN_STORAGE_BITS,
+		.storagebits = CHN_STORAGE_BITS,
+		.shift = 0,
+		.is_big_endian = true,
+	},
+	{
+		.sign = 's',
+		.realbits = CHN_STORAGE_BITS,
+		.storagebits = CHN_STORAGE_BITS,
+		.shift = 0,
+		.is_big_endian = true,
+	},
+	{
+		.sign = 'u',
+		.realbits = CHN_STORAGE_BITS,
+		.storagebits = CHN_STORAGE_BITS,
+		.shift = 0,
+		.is_big_endian = false,
```

Comment:
Do these channels have different polarity?

Reply:
yes. this is a output channel that represents the  excitation frequency we write on a register, and its all passed in little endian from the attributes.